src/org/zaproxy/zap/extension/fuzz/httpfuzzer/processors/tagcreator/HttpFuzzerMessageProcessorTagUIPanel.java

```diff
@@ -0,0 +1,146 @@
+
+        extractByRegExFieldRegEx = new JTextField();
+        addLabelWithTextField(extractByRegExFieldRegEx, EXTRACT_BY_REGEX_LABEL_REGEX);
+    }
+
+    private void createMatchByRegExControl(){
+        matchByRegExRadio = new JRadioButton();
+        addRadioButton(matchByRegExRadio, MATCH_BY_REGEX_NAME, true);
+
+        matchByRegExFieldRegEx = new JTextField();
+        addLabelWithTextField(matchByRegExFieldRegEx, MATCH_BY_REGEX_LABEL_REGEX);
+
+        matchByRegExFieldTag = new JTextField();
+        addLabelWithTextField(matchByRegExFieldTag, MATCH_BY_REGEX_LABEL_TAG);
+    }
+
+    private void addRadioButton(JRadioButton radio, String name, Boolean selected) {
+        radio.setText(name);
+        radio.setSelected(selected);
+        fieldsPanel.add(radio);
+        fieldsPanel.add(new Panel());
+        radioGroup.add(radio);
+    }
+
+    private void addLabelWithTextField(JTextField textField, String labelName){
+        JLabel label = new JLabel(labelName,  JLabel.TRAILING);
+        fieldsPanel.add(label);
+        label.setLabelFor(textField);
+        fieldsPanel.add(textField);
+    }
+
+    @Override
+    public JPanel getComponent() {
+        return fieldsPanel;
+    }
+
+    @Override
+    public void setFuzzerMessageProcessorUI(HttpFuzzerMessageProcessorTagUI processorUI) {
+        ITagRule rule = processorUI.getTagRule();
+        if(rule.getClass() == MatchByRegExTagRule.class){
+            MatchByRegExTagRule tagRule = (MatchByRegExTagRule)rule;
+            matchByRegExRadio.setSelected(true);
+            extractByRegExRadio.setSelected(false);
+            matchByRegExFieldRegEx.setText(tagRule.getRegExp());
+            matchByRegExFieldTag.setText(tagRule.getTag());
+        }
+        else if(rule.getClass() == ExtractByRegExTagRule.class){
+            ExtractByRegExTagRule tagRule = (ExtractByRegExTagRule)rule;
+            matchByRegExRadio.setSelected(false);
+            extractByRegExRadio.setSelected(true);
+            extractByRegExFieldRegEx.setText(tagRule.getRegExp());
+        }
+    }
+
+    @Override
+    public HttpFuzzerMessageProcessorTagUI getFuzzerMessageProcessorUI() {        ;
+        ITagRule tagRule = null;
+        if(matchByRegExRadio.isSelected()){
+            tagRule = new MatchByRegExTagRule(
+                    matchByRegExFieldRegEx.getText(),
+                    matchByRegExFieldTag.getText());
+
+        } else if(extractByRegExRadio.isSelected()){
+            tagRule = new ExtractByRegExTagRule(extractByRegExFieldRegEx.getText());
+        }
+        return new HttpFuzzerMessageProcessorTagUI(tagRule);
+    }
+
+    @Override
+    public void clear() {
+        matchByRegExRadio.setSelected(true);
+        extractByRegExRadio.setSelected(false);
+        matchByRegExFieldRegEx.setText("");
+        matchByRegExFieldTag.setText("");
+        extractByRegExFieldRegEx.setText("");
+    }
+
+    @Override
+    public boolean validate() {
+        return true;
```

Comment:
It should be validated that the regular expressions are valid (and the user informed if not).
Otherwise, the fuzz process fails if the user introduces a malformed regex.
It might worth validate that all required fields were added too.